pkg/engine/renderer/parser.go

```diff
@@ -14,7 +14,7 @@ import (
 // If the type is not known in the scheme, it tries to parse it as Unstructured
 // TODO(av) could we use something else than a global scheme here? Should we somehow inject it?
 func YamlToObject(yaml string) (objs []runtime.Object, err error) {
-	sepYamlfiles := strings.Split(yaml, "---")
+	sepYamlfiles := strings.Split(yaml, "\n---\n")
```

Comment:
I'm wondering why _we_ have to split the file here. Isn't this something the YAML decoder should take care of? Looking at the [documentation of the decoder](https://godoc.org/k8s.io/apimachinery/pkg/util/yaml) it looks like this is a supported use case. The `Decode` functions unmarshals the "next object from the underlying stream". I didn't test this though, but maybe a for-loop that calls `Decode` would do the splitting?

Not yours, but also wondering why we're using a `YAMLOrJSONDecoder` when there's a `YAMLDecoder`?

Reply:
What Jan said ^^ There is a [splitYamlDocument](https://github.com/kubernetes/apimachinery/blob/a98ff070d70e1d5c58428a86787e7a05a38cabe8/pkg/util/yaml/decoder.go#L142) method that should handle multiple YAML documents. Funny enough, the separator it uses is defined as `const yamlSeparator = "\n---"` so only one `\n`.

Reply:
> why we're using a YAMLOrJSONDecoder when there's a YAMLDecoder?

Turns out `YAMLDecoder` is not a decoder, it's a `ReadCloser` :laughing: 
Or at least not in the sense that YAMLOrJSONDecoder provides a `Decode` function.